tools/build.gradle.kts

```diff
@@ -17,19 +19,58 @@
 plugins {
     id("application")
     id("com.hedera.block.tools")
+    id("com.gradleup.shadow") version "8.3.5"
 }
 
 description = "Hedera Block Stream Tools"
 
 application {
-    mainModule = "com.hedera.block.tools"
     mainClass = "com.hedera.block.tools.BlockStreamTool"
 }
 
+// Generate Manifest with Main-Class and Implementation-Title
+tasks.withType<Jar> {
+    manifest {
+        attributes(
+            "Main-Class" to application.mainClass.get(),
+            "Implementation-Title" to project.name,
+            "Implementation-Version" to project.version
+        )
+    }
+}
+
+// Switch compilation from modules back to classpath because 3rd party libraries are not modularized
+tasks.compileJava {
+    // Do not use '--module-path' despite `module-info.java`
+    modularity.inferModulePath = false
+    // Do not compile module-info, we use it only to extract dependencies for now
+    exclude("module-info.java")
+}
+
+// Allow non-module Jar
+extraJavaModuleInfo {
+    failOnMissingModuleInfo = false
+    failOnAutomaticModules = false
+}
+
+// Disable module directives scope check as we are not using modules
+tasks.withType<ModuleDirectivesScopeCheck>().configureEach {
+    enabled = false
+}
```

Comment:
Note: after updating to the new Gradle configuration (#216) I will take another look at how we could improve this to have the check configured without modules.